### PR TITLE
Use Android llm benchmark runner

### PR DESCRIPTION
**Pull request by huydhn**
The current solution is to:

1. Run the benchmark activity
2. Poll the result for up to 10m as 1 is async as @kirklandsign points out.  This looks like an arbitrary threshold for me, but it works so I think we can keep it for now until a better solution comes along
3. Store the JSON result as a local artifact so that we can download it later.  I will have a follow-up PR to propose its format.  I need to cat the result here instead of pulling the file from device as the latter ends up with permission error

### Testing

https://github.com/pytorch/executorch/actions/runs/10731052861/job/29761525913

Download the artifacts from AWS and confirm that the `benchmark_results.json` file are there together with the `instrument.log`

```
{"generateEnd":1725590151329,"generateStart":1725590150258,"loadEnd":1725590150258,"loadStart":1725590149993,"tokens":"tokens/second: 115.530304"}
```

### CODE REVIEW

File: examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml [hunk 1]
```diff
@@ -73,8 +73,16 @@ phases:
           fi
         fi;
 
+      # Run the new generic benchmark activity
+      - echo "Run LLM benchmark"
+      - |
+        adb -s $DEVICEFARM_DEVICE_UDID shell am start -n com.example.executorchllamademo/.Benchmarking \
+        --es "model_dir" "/data/local/tmp/llama" \
+        --es "tokenizer_path" "/data/local/tmp/llama/tokenizer.bin"
```
Comment (kirklandsign) [inline examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml:81]: Thank you! Exactly what i want to do :D 

Need to find a way to wait for the file to result file to appear. shell am start is async

Comment (huydhn) [reply examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml:81]: Oh, got it. TIL.  I'm still working on this to make the command works, so stay tuned :)

Comment (kirklandsign) [reply examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml:81]: Does this kind of stuff work? 🤔 
adb shell while [ ! -f /data/local/tmp/result.txt ]; do sleep 1; done

**Comment by kirklandsign** [comment]
Please format 🥲  https://github.com/google/google-java-format/releases download the binary and run... sorry no built in tool right now

File: examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml [hunk 2]
```diff
@@ -73,16 +73,29 @@ phases:
...
 
   post_test:
     commands:
-      - adb -s $DEVICEFARM_DEVICE_UDID shell "ls -la /data/local/tmp/llama/"
+      - echo "Wait for the results"
+      - |
+        # TODO (huydhn): Polling like this looks brittle, figure out if there is a better way to wait
+        # for the benchmark results
+        adb -s $DEVICEFARM_DEVICE_UDID shell run-as com.example.executorchllamademo \
+        while ! test -f files/benchmark_results.json; do echo "Waiting for benchmark results..."; sleep 30; done
```
Comment (kirklandsign) [inline examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml:90]: Can we have a maximum timeout? Or just rely on GH to timeout?

File: examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml [hunk 3]
```diff
@@ -73,16 +73,29 @@ phases:
...
+        adb -s $DEVICEFARM_DEVICE_UDID shell am start -W -n com.example.executorchllamademo/.LlmBenchmarkRunner \
         --es "model_dir" "/data/local/tmp/llama" \
         --es "tokenizer_path" "/data/local/tmp/llama/tokenizer.bin"
 
   post_test:
     commands:
-      - adb -s $DEVICEFARM_DEVICE_UDID shell "ls -la /data/local/tmp/llama/"
+      - echo "Wait for the results"
+      - |
+        # TODO (huydhn): Polling like this looks brittle, figure out if there is a better way to wait
```
Comment (kirklandsign) [inline examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml:87]: Curious what could be brittle with this approach?

File: examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml (hunk 2 above)
Comment (huydhn) [reply examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml:90]: Yup, we can have a maximum timeout, as GH action timeout is 1 hour which is a bit too long.

File: examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml (hunk 3 above)
Comment (huydhn) [reply examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml:87]: I just feel that there might be a better approach out there, so put a TODO here to remind myself for now

File: examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml (hunk 1 above)
Comment (huydhn) [reply examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml:81]: Yeah, it's working, but adb shell doesn't like the way I write bash script, so I need to look for a work around by cat-ing the results.  It works nonetheless, so I guess we are good :)

**Review by kirklandsign — APPROVED** [review]

**Pull request merged**